src/earn/prepareTransactions.ts

```diff
@@ -112,106 +152,125 @@ export async function prepareWithdrawAndClaimTransactions({
       tokens: [
         {
           tokenId: dataProps.withdrawTokenId,
-          amount: useMax ? balance : amount,
-          useMax,
+          amount: balance,
+          useMax: true,
         },
       ],
       ...shortcutTriggerArgs?.withdraw,
     })
+
   const claimTransactions = await Promise.all(
     rewardsPositions.map(async (position): Promise<RawShortcutTransaction[]> => {
-      const { transactions }: { transactions: RawShortcutTransaction[] } =
+      const { transactions }: { transactions?: RawShortcutTransaction[] } =
         await triggerShortcutRequest(hooksApiUrl, {
           address: walletAddress,
           appId,
           networkId,
           shortcutId: 'claim-rewards',
           ...position.shortcutTriggerArgs?.['claim-rewards'],
         })
-      return transactions
+      return transactions ?? [] // Default to an empty array if rewardsPositions is undefined
     })
   )
   Logger.debug(TAG, 'prepareWithdrawAndClaimTransactions', {
     withdrawTransactions,
     claimTransactions,
     pool,
   })
+
   return prepareTransactions({
     feeCurrencies,
     baseTransactions: rawShortcutTransactionsToTransactionRequests([
       ...withdrawTransactions,
-      ..._.flatten(claimTransactions),
+      ...(dataProps.withdrawalIncludesClaim ? [] : claimTransactions.flat()),
```

Comment:
Do we even need to call the `'claim-rewards'` shortcut if withdrawalIncludesClaim === true ?

Reply:
Also please add a test for this added logic

Reply:
Calling `claim-rewards` if withdrawalIncludesClaim === true allows the users to claim-rewards without withdrawing.

Reply:
That's a good point; we don't need to call `'claim-rewards'` here. Code change a65343a36aa1609ff4ae4008a1936b92cc361f4e and test 47787a8b084e759a52ca1d9d2bd103cbab2f43ac.